apps/api/src/app/rate-limiting/usecases/evaluate-api-rate-limit/evaluate-api-rate-limit.usecase.ts

```diff
@@ -20,13 +21,24 @@ export class EvaluateApiRateLimit {
 
   @InstrumentUsecase()
   async execute(command: EvaluateApiRateLimitCommand): Promise<EvaluateApiRateLimitResponseDto> {
-    const [maxLimitPerSecond, apiServiceLevel] = await this.getApiRateLimitMaximum.execute(
-      GetApiRateLimitMaximumCommand.create({
-        apiRateLimitCategory: command.apiRateLimitCategory,
-        environmentId: command.environmentId,
-        organizationId: command.organizationId,
-      })
-    );
+    let maxLimitPerSecond: number;
+    let apiServiceLevel: ApiServiceLevel;
+
+    // For keyless environments, we implement strict rate limiting to prevent abuse:
+    if (!command.organizationId || !command.environmentId) {
+      maxLimitPerSecond = 3000;
+      apiServiceLevel = ApiServiceLevelEnum.FREE;
+    } else {
+      const result = await this.getApiRateLimitMaximum.execute(
```

Comment:
```suggestion
     [maxLimitPerSecond, apiServiceLevel] = await this.getApiRateLimitMaximum.execute(
```

I believve this will work too

Reply:
yes it does 😃